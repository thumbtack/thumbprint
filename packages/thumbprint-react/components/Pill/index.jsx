import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import styles from './index.module.scss';

export default function Pill({ color, icon, children }) {
    return (
        <div
            className={classNames({
                [styles.root]: true,
                [styles.pillGreen]: color === 'green',
                [styles.pillRed]: color === 'red',
                [styles.pillIndigo]: color === 'indigo',
                [styles.pillBlue]: color === 'blue',
                [styles.pillYellow]: color === 'yellow',
                [styles.pillPurple]: color === 'purple',
            })}
        >
            <div className={styles.content}>
                {icon && <span className={styles.iconWrap}>{icon}</span>}
                <span>{children}</span>
            </div>
        </div>
    );
}

Pill.propTypes = {
    /**
     * Text content to render.
     */
    children: PropTypes.string.isRequired,
    /**
     * "Tiny" size icon to render.
     */
    icon: PropTypes.node,
    /**
     * Color of the pill text and background.
     */
    color: PropTypes.oneOf(['green', 'red', 'indigo', 'blue', 'yellow', 'purple']),
};

Pill.defaultProps = {
    icon: undefined,
    color: undefined,
};
