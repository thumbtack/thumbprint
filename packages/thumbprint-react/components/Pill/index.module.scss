@import '~@thumbtack/thumbprint-tokens/dist/scss/_index';

.root {
    height: 24px;
    padding-left: 12px;
    padding-right: 12px;
    border-radius: 12px;
    background: $tp-color__gray-300;
    color: $tp-color__black;
    display: inline-block;
    font-size: $tp-font__body__2__size;
    line-height: 23px; // Non-standard line-height to properly vertical align.
    font-weight: $tp-font-weight__bold;
    overflow: hidden;
}

.iconWrap {
    width: 14px;
    height: 14px;
    overflow: hidden;
    margin-right: $tp-space__1;

    svg {
        display: block;
    }
}

.content {
    display: flex;
    align-items: center;
}

.pillGreen {
    background: $tp-color__green-100;
    color: $tp-color__green-600;
}

.pillRed {
    background: $tp-color__red-100;
    color: $tp-color__red-600;
}

.pillIndigo {
    background: $tp-color__indigo-100;
    color: $tp-color__indigo-600;
}

.pillBlue {
    background: $tp-color__blue-100;
    color: $tp-color__blue-600;
}

.pillYellow {
    background: $tp-color__yellow-100;
    color: $tp-color__yellow-600;
}

.pillPurple {
    background: $tp-color__purple-100;
    color: $tp-color__purple-600;
}
