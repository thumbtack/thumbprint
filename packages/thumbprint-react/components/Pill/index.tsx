import React from 'react';
import classNames from 'classnames';

import styles from './index.module.scss';

interface PropTypes {
    /**
     * Text content to render.
     */
    children: string;
    /**
     * "Tiny" size icon to render.
     */
    icon?: React.ReactNode;
    /**
     * Color of the pill text and background.
     */
    color?: 'green' | 'red' | 'indigo' | 'blue' | 'yellow' | 'purple';
}

export default function Pill({ color, icon, children }: PropTypes): JSX.Element {
    return (
        <div
            className={classNames({
                [styles.root]: true,
                [styles.pillGreen]: color === 'green',
                [styles.pillRed]: color === 'red',
                [styles.pillIndigo]: color === 'indigo',
                [styles.pillBlue]: color === 'blue',
                [styles.pillYellow]: color === 'yellow',
                [styles.pillPurple]: color === 'purple',
            })}
        >
            <div className={styles.content}>
                {icon && <span className={styles.iconWrap}>{icon}</span>}
                <span>{children}</span>
            </div>
        </div>
    );
}
