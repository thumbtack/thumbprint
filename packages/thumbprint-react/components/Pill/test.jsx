import React from 'react';
import { mount } from 'enzyme';
import { ContentModifierLightningTiny } from '@thumbtack/thumbprint-icons';
import Pill from './index';

test('renders content that is passed in', () => {
    const wrapper = mount(<Pill>goose</Pill>);
    expect(wrapper.text()).toBe('goose');
    expect(wrapper).toMatchSnapshot();
});

test('renders icon', () => {
    const wrapper = mount(<Pill icon={<ContentModifierLightningTiny />}>goose</Pill>);
    expect(wrapper.find('svg')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
});

test('sets color theme', () => {
    const wrapper = mount(<Pill color="red">goose</Pill>);
    expect(wrapper.find('.root').hasClass('pillRed'));
    expect(wrapper).toMatchSnapshot();
});
