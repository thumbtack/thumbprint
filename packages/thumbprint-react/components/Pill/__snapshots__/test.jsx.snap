// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content that is passed in 1`] = `
<Pill>
  <div
    className="root"
  >
    <div
      className="content"
    >
      <span>
        goose
      </span>
    </div>
  </div>
</Pill>
`;

exports[`renders icon 1`] = `
<Pill
  icon={
    <ContentModifierLightningTiny
      fill="currentColor"
      height="14"
      viewBox="0 0 14 14"
      width="14"
      xmlns="http://www.w3.org/2000/svg"
    />
  }
>
  <div
    className="root"
  >
    <div
      className="content"
    >
      <span
        className="iconWrap"
      >
        <ContentModifierLightningTiny
          fill="currentColor"
          height="14"
          viewBox="0 0 14 14"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <svg
            fill="currentColor"
            height="14"
            viewBox="0 0 14 14"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.968 10.08l.897-2.855a.751.751 0 0 0-.715-.975H4.14L6.604 2.5h3.019L8.101 5.123A.748.748 0 0 0 8.75 6.25h.898l-4.68 3.83zm7.488-4.832a.748.748 0 0 0-.706-.498h-1.698l1.523-2.623A.752.752 0 0 0 10.925 1H6.201a.75.75 0 0 0-.628.338l-3.45 5.25A.75.75 0 0 0 2.75 7.75h1.378l-1.344 4.275a.749.749 0 0 0 1.19.806l8.25-6.75a.749.749 0 0 0 .232-.833z"
            />
          </svg>
        </ContentModifierLightningTiny>
      </span>
      <span>
        goose
      </span>
    </div>
  </div>
</Pill>
`;

exports[`sets color theme 1`] = `
<Pill
  color="red"
>
  <div
    className="root pillRed"
  >
    <div
      className="content"
    >
      <span>
        goose
      </span>
    </div>
  </div>
</Pill>
`;
